src/globus_sdk/transport/retry_policy.py

```diff
@@ -0,0 +1,49 @@
+import time
+import typing
+
+import requests
+
+
+class RetryPolicy:
+    max_retries: int = 5
+    max_sleep: int = 10
+
+    def _backoff(self, attempt):
+        # (0.5s * 2^attempt) results in ~15s of total sleep for 5 retries
+        # 2^0 + 2^1 + 2^2 + 2^3 + 2^4 = 31
+        # note that additional time can be spent sending the request or waiting for
+        # a reply
+        time.sleep(0.5 * (2 ** attempt))  # exponential backoff
+
+    def _should_retry_err(self, attempt: int, error: Exception) -> bool:
+        if isinstance(error, requests.RequestException):
+            self._backoff(attempt)
+            return True
+        return False
+
+    def _parse_retry_after(self, response: requests.Response) -> typing.Optional[int]:
+        val = response.headers.get("Retry-After")
+        if val:
+            try:
+                return int(val)
+            except ValueError:
+                pass
+        return None
```

Comment:
I think this would have the same behavior?
```suggestion
        try:
            return int(response.headers.get("Retry-After"))
        except TypeError, ValueError:
            pass
```